### DISCUSSION TITLE
How to support customized topic message  type in 3D pannel?

### DISCUSSION THREAD

**Discussion by Elaine87** [foxglove/community#73] [Q&A]
My topic type is  livox_ros_driver/CustomMsg.  livox_ros_driver/CustomMsg data package format, as follows :

`std_msgs/Header header
  uint32 seq
  time stamp
  string frame_id
uint64 timebase
uint32 point_num
uint8 lidar_id
uint8[3] rsvd
livox_ros_driver/CustomPoint[] points
  uint32 offset_time
  float32 x
  float32 y
  float32 z
  uint8 reflectivity
  uint8 tag
  uint8 line
`
![image](https://user-images.githubusercontent.com/3049861/149468081-61920a80-8b76-4599-98b3-ccef2957d8e7.png)

I want to show local Ros1 bag file(eg: lidar0_2021-05-23-16-00-41_0.bag has topic  type livox_ros_driver/CustomMsg)  in 3D pannel.

Thanks!

**Comment by defunctzombie** [comment]
To show messages in the 3d panel, the messages must be in one of the supported datatypes. These are documented in the _Supported display types_ section of our 3d panel docs: https://foxglove.dev/docs/panels/3d

If your message is in this datatype, you can use [Node Playground](https://foxglove.dev/docs/panels/node-playground) to write a transformation in Studio to convert from your format into one of the supported formats. You might find it easier to do this conversion on your robot and publish a conforming message.
